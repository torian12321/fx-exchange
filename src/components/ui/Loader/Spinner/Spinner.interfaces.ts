export interface ISpinner {
  children?: never;
  className?: string;
}
  