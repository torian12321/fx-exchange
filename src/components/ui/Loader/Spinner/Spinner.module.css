.spinner {
  color: #b159a4;
  padding: 0;
  border: 4px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  font-size: 24px;
  height: 1em;
  width: 1em;
  background: transparent;

  animation: spin 1s infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
  