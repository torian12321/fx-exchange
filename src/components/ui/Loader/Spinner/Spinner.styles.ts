import { css, keyframes } from '@emotion/core'

const color = '#b159a4'

const spin = keyframes`
  0%   { transform: rotate(0deg);   }
  100% { transform: rotate(360deg); }
`;

export const spinner = css`
  color: ${color};
  padding: 0;
  border: 4px solid currentColor;
  border-top-color: transparent;
  border-radius: 50%;
  font-size: 24px;
  height: 1em;
  width: 1em;
  background: transparent;

  animation: ${spin} 1s infinite;
`;
