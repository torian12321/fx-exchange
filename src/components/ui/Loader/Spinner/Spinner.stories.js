import React from 'react';
import { Spinner } from './Spinner';

export default {
  title: 'UI Components|Loaders/Spinner',
};

export const Basic = () => <Spinner />;
