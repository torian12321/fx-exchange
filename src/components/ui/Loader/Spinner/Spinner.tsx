import React from "react";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import { ISpinner } from './Spinner.interfaces';
import * as styles from "./Spinner.styles";

const Spinner = ({ className }: ISpinner) => (
  <div css={styles.spinner} className={className} />
);

export { Spinner };