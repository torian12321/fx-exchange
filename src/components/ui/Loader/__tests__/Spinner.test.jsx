import React from "react";
import { create } from "react-test-renderer";
import { Spinner } from "../Spinner";

describe("UI Component: Loader -> Spinner", () => {
  it(`should render`, () => {
    const component = create(
      <Spinner />
    );

    const spinner = component.root;

    expect(spinner.findAllByType("div").length).toEqual(1);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
