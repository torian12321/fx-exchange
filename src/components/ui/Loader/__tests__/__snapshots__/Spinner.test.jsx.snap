// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component: Loader -> Spinner should render 1`] = `
<div
  className=" css-93jcja-Spinner"
/>
`;
